app/ui/src/app/integration/integration_detail/integration_activity/integration-activity.component.ts

```diff
@@ -97,6 +99,17 @@ export class IntegrationActivityComponent implements OnInit {
     this.showPagination = (this.allActivities.length > this.paginationConfig.pageSize);
     this.paginationConfig.totalItems = this.allActivities.length;
     this.paginationConfig.pageNumber = 1;
+
+    this.openshiftConsoleURL = null;
+    if (this.allActivities.length > 0) {
+      const base = this.configService.getSettings('consoleUrl');
+      const pod = this.allActivities[0].pod;
```

Comment:
Why not just expose the pod ID via the integration object?  This seems a smell...

Reply:
wasn't my code, so have to dig a bit into it. From where do I have access to the pod from the integration object ? I can't see a connection to the pod id from the IntegrationDeployment either (which I probably would have to fetch first then, too)

For me it seems like that only Activity has a reference to a pod. Of course it could be possible that this pod is not alive anymore, but as discussed with @chirino and @kcbabo  it seems to be acceptable that the generated link might be a stale one leading to 404 (e.g. when the pod has been replaced or the integration is not running anymore) when selected.

Reply:
yeah, that's why I'm saying this is a smell :-)  The only reference to a pod is in what is effectively a log statement?  If we need the pod ID then we should make it available to the integration object.

Reply:
An integration runs on multiple pods over time.  An integration object only changes when a user edits it.  It would be a smell to change it due to runtime state.